src/k8s/pkg/k8sd/api/cluster_config.go

```diff
@@ -27,12 +27,19 @@ func (e *Endpoints) putClusterConfig(s *state.State, r *http.Request) response.R
 		return response.BadRequest(fmt.Errorf("failed to decode request: %w", err))
 	}
 
+	// Changing the datastore configuration is opt-in. We expect the caller to explicitly set the "external" type.
+	// The nil check is required to ensure we only fail if the DatastoreConfig is expected to change.
+	if req.DatastoreConfig.Type != nil && req.DatastoreConfig.GetType() != "external" {
+		return response.BadRequest(fmt.Errorf("failed to updated datastore config: type must be 'external' but is %s", req.DatastoreConfig.GetType()))
```

Comment:
let's use `%q` for quotes, as it also handles escapes for us

```suggestion
		return response.BadRequest(fmt.Errorf("failed to updated datastore config: type must be %q but is %q", "external", req.DatastoreConfig.GetType()))
```